pkg/relay/relay.go

```diff
@@ -145,6 +164,26 @@ func NewRelay(l log.Logger, config RelayConfig, beacon Beacon, cache ValidatorCa
 	return rs
 }
 
+func (rs *Relay) RunSlotDeliveredUpdater(ctx context.Context) error {
+	for {
+		select {
+		case slot := <-rs.s.SlotDeliveredChan():
+			rs.updateSlotDelivered(slot)
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (rs *Relay) updateSlotDelivered(slot structs.Slot) {
+	rs.slotMu.Lock()
+	defer rs.slotMu.Unlock()
+
+	if rs.slotDelivered < slot {
+		rs.slotDelivered = slot
+	}
+}
```

Comment:
`slotDelivered` is just an integer I don't thing we need to lock here. Especially that  `updateSlotDelivered()` is triggered by select so is a synchronous operation. 
I think  this should be enough:
```go 

select {
		case slot := <-rs.s.SlotDeliveredChan():
                      if rs.slotDelivered < slot {
        		rs.slotDelivered = slot
	             }
		case <-ctx.Done():
			return ctx.Err()
		}
```